cdap-ui/app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsButtons/ScheduleButton.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React, { Component } from 'react';
+import PipelineDetailStore from 'components/PipelineDetails/store';
+import PipelineScheduler from 'components/PipelineScheduler';
+import classnames from 'classnames';
+import IconSVG from 'components/IconSVG';
+import StatusMapper from 'services/StatusMapper';
+
+export default class ScheduleButton extends Component {
+  state = {
+    showScheduler: false,
+    scheduleStatus: "",
+    scheduleLoading: true
+  };
+
+  detailStoreSubscription = PipelineDetailStore.subscribe(() => {
+    let scheduleStatus = StatusMapper.lookupDisplayStatus(PipelineDetailStore.getState().scheduleStatus);
+    if (scheduleStatus !== this.state.scheduleStatus) {
+      this.setState({
+        scheduleStatus,
+        scheduleLoading: false
+      });
+    }
+  });
+
+  componentWillUnmount() {
+    if (this.detailStoreSubscription) {
+      this.detailStoreSubscription();
+    }
+  }
+
+  toggleScheduler = () => {
+    this.setState({
+      showScheduler: !this.state.showScheduler
+    });
+  }
+
+  renderSchedulerButton() {
+    if (['Deployed', 'Scheduling'].indexOf(this.state.scheduleStatus) !== -1 || this.state.scheduleStatus.error) {
```

Comment:
Might want to use constant `statusMap` from `StatusMapper`. Would be hard to track when we change it there.